### PR TITLE
[kubeadm] Use KubeletRunDirectory instead of hard-coded way

### PR DESCRIPTION
**Pull request by xlgao-zju**
**What this PR does / why we need it**:
fix `TODO: Make hard-coded "/var/lib/kubelet" strings reference this constant.`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
cc @neolit123 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/reset.go [hunk 1]
```diff
@@ -138,21 +138,21 @@ func (r *Reset) Run(out io.Writer) error {
...
-	umountDirsCmd := "awk '$2 ~ path {print $2}' path=/var/lib/kubelet /proc/mounts | xargs -r umount"
+	// Try to unmount mounted directories under kubeadmconstants.KubeletRunDirectory in order to be able to remove the kubeadmconstants.KubeletRunDirectory directory later
+	fmt.Printf("[reset] unmounting mounted directories in %q\n", kubeadmconstants.KubeletRunDirectory)
+	umountDirsCmd := fmt.Sprintf("awk '$2 ~ path {print $2}' path=%q /proc/mounts | xargs -r umount", kubeadmconstants.KubeletRunDirectory)
 
 	glog.V(1).Infof("[reset] executing command %q", umountDirsCmd)
 	umountOutputBytes, err := exec.Command("sh", "-c", umountDirsCmd).Output()
 	if err != nil {
-		glog.Errorf("[reset] failed to unmount mounted directories in /var/lib/kubelet: %s\n", string(umountOutputBytes))
+		glog.Errorf("[reset] failed to unmount mounted directories in %q: %q\n", kubeadmconstants.KubeletRunDirectory, string(umountOutputBytes))
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/reset.go]: `%q: %q` -> `%q: %s`

**Comment by neolit123** [comment]
@xlgao-zju 
thanks for the PR.

there is another usage of `/var/lib/kubelet` in `app/phases/kubelet/flags_test.go`. i think we should update that one too?
makes sure to run `./hack/update-bazel.sh` once you include `cmd/kubeadm/app/constants/` in that file.

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews

**Review by neolit123 — COMMENTED** [review]
one small nit.

File: cmd/kubeadm/app/cmd/reset.go [hunk 2]
```diff
@@ -138,21 +138,21 @@ func (r *Reset) Run(out io.Writer) error {
 		}
 	}
 
-	// Try to unmount mounted directories under /var/lib/kubelet in order to be able to remove the /var/lib/kubelet directory later
-	fmt.Printf("[reset] unmounting mounted directories in %q\n", "/var/lib/kubelet")
-	umountDirsCmd := "awk '$2 ~ path {print $2}' path=/var/lib/kubelet /proc/mounts | xargs -r umount"
+	// Try to unmount mounted directories under kubeadmconstants.KubeletRunDirectory in order to be able to remove the kubeadmconstants.KubeletRunDirectory directory later
+	fmt.Printf("[reset] unmounting mounted directories in %q\n", kubeadmconstants.KubeletRunDirectory)
+	umountDirsCmd := fmt.Sprintf("awk '$2 ~ path {print $2}' path=%q /proc/mounts | xargs -r umount", kubeadmconstants.KubeletRunDirectory)
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/reset.go]: must be `%s` here

File: cmd/kubeadm/app/cmd/reset.go (hunk 1 above)
Comment (luxas) [reply cmd/kubeadm/app/cmd/reset.go]: I'd do `%s: %s` which would preserve the current behavior :smile:

**Comment by neolit123** [comment]
@xlgao-zju

just change these as @luxas wants them.

> I thought we will always use %q, when it is a string. When should we use %s? Is there a codebase which explain when we should use what?

so in terms of golang, `%q` double quotes the string - e.g. `some-string`, becomes `"some-string"`.
if you don't want to double quote a string `%s` should be used.

it can depend on how a sentence is formatted:
```
someComand := "test"
fmt.Printf("the output of the command %q was correct", someCommand");
```
```
the output of the command "test" was correct
```
vs
```
someComand := "test"
fmt.Printf("the output of the following command was correct: %s", someCommand");
```
```
the output of the following command was correct: test
```

but also `%q` vs `%s` can break things if quotes are wanted or unwanted on the command line.
hope that explains.

**Comment by xlgao-zju** [comment]
@neolit123 Thank you for your explanation, so kind you are. 😆 

> make sure to run ./hack/update-bazel.sh once you include cmd/kubeadm/app/constants/ in that file.

What does this `update-bazel.sh` do? Updating the generated files? I am not quite family with this... 😓

**Comment by neolit123** [comment]
> What does this update-bazel.sh do? Updating the generated files? I am not quite family with this...

yes, it updates the generated files for bazel - files called `BUILD`. make sure to `git add ...` when making a commit.
https://github.com/bazelbuild/bazel/blob/master/BUILD

**Comment by xlgao-zju** [comment]
> make sure to run ./hack/update-bazel.sh once you include cmd/kubeadm/app/constants/ in that file.

@neolit123 Is there other condition I should run `./hack/update-bazel.sh`?

**Comment by neolit123** [comment]
> Is there other condition I should run ./hack/update-bazel.sh?

when creating new files / packages too.

**Comment by xlgao-zju** [comment]
@neolit123 Thank you for all your help. 😆  
This PR is updated. PTAL

**Comment by neolit123** [comment]
/lgtm
thanks @xlgao-zju

**Review by luxas — APPROVED** [review]
/approve
Thanks for the contribution @xlgao-zju!

**Pull request merged**